code/core/src/test/java/com/adobe/marketing/mobile/util/EventUtilsTest.kt

```diff
@@ -0,0 +1,189 @@
+                mapOf(
+                    "debug" to mapOf(
+                        "eventType" to EventType.RULES_ENGINE,
+                        "eventSource" to EventSource.RESET_COMPLETE
+                    )
+                )
+            ).build()
+
+        assertEquals(EventType.RULES_ENGINE, event.getDebugEventType())
+    }
+
+    @Test
+    fun `Test getDebugEventType returns null when debugEvent type is not a string`() {
+        val event = Event.Builder(TEST_EVENT_NAME, TEST_EVENT_TYPE, TEST_EVENT_SOURCE)
+            .setEventData(
+                mapOf(
+                    "debug" to mapOf(
+                        "eventType" to mapOf("1" to "Hi"),
+                        "eventSource" to "testEventSource"
+                    )
+                )
+            ).build()
+
+        assertNull(event.getDebugEventType())
+    }
+
+    @Test
+    fun `Test getDebugEventSource returns null on non debug event`() {
+        val event = Event.Builder(TEST_EVENT_NAME, EventType.HUB, EventSource.REQUEST_CONTENT).build()
+
+        assertNull(event.getDebugEventSource())
+    }
+
+    @Test
+    fun `Test getDebugEventSource returns null on no eventData`() {
+        val event = Event.Builder(TEST_EVENT_NAME, TEST_EVENT_TYPE, TEST_EVENT_SOURCE).build()
+
+        assertNull(event.getDebugEventSource())
+    }
+
+    @Test
+    fun `Test getDebugEventSource returns null on no debug data`() {
+        val event = Event.Builder(TEST_EVENT_NAME, TEST_EVENT_TYPE, TEST_EVENT_SOURCE)
+            .setEventData(mapOf("key" to "value")).build()
+
+        assertNull(event.getDebugEventSource())
+    }
+
+    @Test
+    fun `Test getDebugEventSource returns null on invalid debug key`() {
+        val event = Event.Builder(TEST_EVENT_NAME, TEST_EVENT_TYPE, TEST_EVENT_SOURCE)
+            .setEventData(
+                mapOf(
+                    "_debug" to mapOf(
+                        "eventType" to EventType.RULES_ENGINE,
+                        "eventSource" to EventSource.RESET_COMPLETE
+                    )
+                )
+            ).build()
+
+        assertNull(event.getDebugEventSource())
+    }
+
+    @Test
+    fun `Test getDebugEventSource returns debug event source`() {
+        val event = Event.Builder(TEST_EVENT_NAME, TEST_EVENT_TYPE, TEST_EVENT_SOURCE)
+            .setEventData(
+                mapOf(
+                    "debug" to mapOf(
+                        "eventType" to EventType.RULES_ENGINE,
+                        "eventSource" to EventSource.RESET_COMPLETE
+                    )
+                )
+            ).build()
+
+        assertEquals(EventSource.RESET_COMPLETE, event.getDebugEventSource())
+    }
+
+    @Test
+    fun `Test getDebugEventSource returns null when debugEvent source is not a string`() {
+        val event = Event.Builder(TEST_EVENT_NAME, TEST_EVENT_TYPE, TEST_EVENT_SOURCE)
+            .setEventData(
+                mapOf(
+                    "debug" to mapOf(
```

Comment:
Can you add a test where the value of debug is not a map?

Reply:
Done!